nixos/modules/services/cluster/rke2/default.nix

```diff
@@ -0,0 +1,276 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.rke2;
+in
+{
+  imports = [];
+
+  options.services.rke2 = {
+    enable = mkEnableOption "rke2";
+
+    package = mkPackageOption pkgs "rke2" {};
+
+    role = mkOption {
+      type = types.enum [ "server" "agent" ];
+      description = ''
+        Whether rke2 should run as a server or agent.
+
+        If it's a server:
+
+        - By default it also runs workloads as an agent.
+        - any optionals is allowed.
+
+        If it's an agent:
+
+        - `serverAddr` is required.
+        - `token` or `tokenFile` is required.
+        - `agentToken` or `agentTokenFile` or `disable` or `cni` are not allowed.
+      '';
+      default = "server";
+    };
+
+    configPath = mkOption {
+      type = types.path;
+      description = "Load configuration from FILE.";
+      default = "/etc/rancher/rke2/config.yaml";
+    };
+
+    debug = mkOption {
+      type = types.bool;
+      description = "Turn on debug logs.";
+      default = false;
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      description = "The folder to hold state in.";
+      default = "/var/lib/rancher/rke2";
+    };
+
+    token = mkOption {
+      type = types.str;
+      description = ''
+        Shared secret used to join a server or agent to a cluster.
+
+        > WARNING: This option will expose store your token unencrypted world-readable in the nix store.
+        If this is undesired use the `tokenFile` option instead.
+      '';
+      default = "";
+    };
+
+    tokenFile = mkOption {
+      type = types.nullOr types.path;
+      description = "File path containing rke2 token to use when connecting to the server.";
+      default = null;
+    };
+
+    disable = mkOption {
+      type = with types;
+        listOf (enum [ "rke2-coredns" "rke2-ingress-nginx" "rke2-metrics-server" ]);
```

Comment:
If this works similar to the disable option in k3s, which it does according to the [docs](https://docs.rke2.io/helm#disabling-addons), you can disable not only built-in add-ons but also any arbitrary manifest from the manifest directory. This shouldn't be an enum but allow any non-empty string.

Reply:
I don't rule out that more will be added in the future, but according to this [docs](https://docs.rke2.io/reference/server_config#components) (and my experiment), the default startup of AddOns are just `rke2-coredns`, `rke2-ingress-nginx` and `rke2-metrics-server`.

Reply:
Yes, that's true. However, if you place a file `my-addon.yaml` in `/var/lib/rancher/rke2/server/manifests/`, the add-on controller will apply every resource in the file. If you now want to let the add-on controller delete the applied resources, you need to pass `--disable my-addon`. This wouldn't be possible with the `disable` option, if it takes an enum.

I guess the important part of the [docs](https://docs.rke2.io/helm#disabling-addons) is 
> in addition to AddOns for any additional manifests placed in the manifests directory